sensor/common/processsignal/pipeline.go

```diff
@@ -52,6 +71,25 @@ func populateIndicatorFromCachedContainer(indicator *storage.ProcessIndicator, c
 	indicator.ImageId = cachedContainer.ImageID
 }
 
+// Shutdown closes all communication channels and shutdowns the enricher
+func (p *Pipeline) Shutdown() {
+	defer close(p.enrichedIndicators)
+	p.ctxCancel()
+}
+
+// Notify allows the component state to be propagated to the pipeline
+func (p *Pipeline) Notify(e common.SensorComponentEvent) {
+	log.Debugf("Received notify: %s", e)
+	switch e {
+	case common.SensorComponentEventCentralReachable:
+		p.centralReady.Signal()
+		log.Debug("ProcessSingnalPipeline runs now in Online mode.")
+	case common.SensorComponentEventOfflineMode:
+		p.centralReady.Reset()
+		log.Debug("ProcessSingnalPipeline runs now in Offline mode.")
```

Comment:
Idea: maybe it would be good to have the same `log.Debug` format message for all components that support online/offline mode. This way we'd also see the order in which components switched to offline mode and vice-versa.

Reply:
I proposed something for this in [063df5a](https://github.com/stackrox/stackrox/pull/7011/commits/063df5a062649af0b2d9f34f6dc390acb4422e35)